test/e2e/node_maintenance_test.go

```diff
@@ -419,6 +422,31 @@ func getOperatorPod() *corev1.Pod {
 	return &pods.Items[0]
 }
 
+func validateOperatorCustomLabels() bool {
+
+	podOperator := getOperatorPod()
+	podName := podOperator.ObjectMeta.Name
+
+	// Get the Pod
+	pod, err := KubeClient.CoreV1().Pods(operatorNsName).Get(context.TODO(), podName, metav1.GetOptions{})
+	//pod, err := KubeClient.CoreV1().Pods(operatorNsName).List(context.Background(), metav1.ListOptions{LabelSelector: "app.kubernetes.io/name="})
+
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to get pod '%s'", podName))
+
+	// Labels to check for
+	requiredKeys := []string{"app.kubernetes.io/name",
+		"app.kubernetes.io/component",
+		"kubectl.kubernetes.io/default-container"}
+
+	// Check if each label is present
+	for _, key := range requiredKeys {
+		_, exists := pod.Labels[key]
+		ExpectWithOffset(1, exists).Should(BeTrue(), fmt.Sprintf("Missing required label '%s' in pod '%s'", key, podName))
```

Comment:
a) you don't check label values anymore
b) hint: ExpectWithOffset(1, pod.Labels).To(HaveKeyWithValue(key, value))